mlflow/pyfunc/__init__.py

```diff
@@ -861,7 +874,26 @@ def load_model(
             .. Note:: Experimental: This parameter may change or be removed in a future
                 release without warning.
     """
-    local_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+
+    lineage_header_info = None
+    if databricks_utils.is_in_databricks_notebook() or databricks_utils.is_in_databricks_job():
+        entity_list = []
+        # Get notebook id and job id, pack them into lineage_header_info
+        notebook_id = databricks_utils.get_notebook_id()
+        if notebook_id:
+            notebook_entity = Notebook(id=str(notebook_id))
+            entity_list.append(Entity(notebook=notebook_entity))
+
+        job_id = databricks_utils.get_job_id()
+        if job_id:
+            job_entity = Job(id=str(job_id))
+            entity_list.append(Entity(job=job_entity))
+
+        lineage_header_info = LineageHeaderInfo(entities=entity_list) if entity_list else None
```

Comment:
@shichengzhou-db could you provide more context about lineage header info ? :)

Reply:
Would you also provide me some example databricks notebook so that I can make some test ? You can send it in slack for privacy.